concatenator/dataset_and_group_handling.py

```diff
@@ -334,6 +334,10 @@ def validate_workable_files(files_to_concat, logger) -> tuple[list[str], int]:
         except OSError:
             logger.debug("Error opening <%s> as a netCDF dataset. Skipping.", file)
 
+    # addressing the issue 153: propagate first empty file if all input files are empty
+    if len(workable_files) == 0:
+        workable_files.append(files_to_concat[0])
```

Comment:
If `files_to_concat` is empty, I think this will raise an error.  Might be better to make that case more explicit, and let the caller handle the empty return from this function.... so, use this instead:

```python
if (len(workable_files) == 0) and (len(files_to_concat) > 0):
```

Reply:
I think the `files_to_concat` in this function should be renamed to `files`, for readability and since the concatenation isn't relevant _within_ this function. @ank1m, does that sound good?  If so, I could change it here, or perhaps you could make that change on your branch?